plugins/volumeFilter_test.go

```diff
@@ -166,6 +166,277 @@ type volumeFilterFnTestCase struct {
+			inputPrice:   2.0,
+			inputAmount:  5.01,
+			wantPrice:    nil,
+			wantAmount:   nil,
+			wantTbbBase:  11,
+			wantTbbQuote: 0,
+		},
+		{
+			name:         "6. tbb = 0; projected < cap",
+			cap:          10.0,
+			otb:          5,
+			tbb:          0,
+			inputPrice:   2.0,
+			inputAmount:  4.99,
+			wantPrice:    pointy.Float64(2.0),
+			wantAmount:   pointy.Float64(4.99),
+			wantTbbBase:  4.99,
+			wantTbbQuote: 9.98,
+		},
+		{
+			name:         "7. tbb = 0; projected = cap",
+			cap:          10.0,
+			otb:          5,
+			tbb:          0,
+			inputPrice:   2.0,
+			inputAmount:  5.0,
+			wantPrice:    pointy.Float64(2.0),
+			wantAmount:   pointy.Float64(5.0),
+			wantTbbBase:  5,
+			wantTbbQuote: 10,
+		},
+		{
+			name:         "8. tbb = 0; projected > cap, newAmount > 0",
+			cap:          10.0,
+			otb:          5,
+			tbb:          0,
+			inputPrice:   2.0,
+			inputAmount:  6.0,
+			wantPrice:    pointy.Float64(2.0),
+			wantAmount:   pointy.Float64(5.0),
+			wantTbbBase:  5,
+			wantTbbQuote: 10,
+		},
+		{
+			name:         "9. tbb = 0; projected > cap, newAmount = 0",
+			cap:          10.0,
+			otb:          10,
+			tbb:          0,
+			inputPrice:   2.0,
+			inputAmount:  6.0,
+			wantPrice:    nil,
+			wantAmount:   nil,
+			wantTbbBase:  0,
+			wantTbbQuote: 0,
+		},
+		{
+			name:         "10. tbb = 0; projected > cap, newAmount < 0",
+			cap:          10.0,
+			otb:          11,
+			tbb:          0,
+			inputPrice:   2.0,
+			inputAmount:  6.0,
+			wantPrice:    nil,
+			wantAmount:   nil,
+			wantTbbBase:  0,
+			wantTbbQuote: 0,
+		},
+		{
+			name:         "11. otb = 0 && tbb = 0; projected < cap",
+			cap:          10.0,
+			otb:          0,
+			tbb:          0,
+			inputPrice:   2.0,
+			inputAmount:  5.0,
+			wantPrice:    pointy.Float64(2.0),
+			wantAmount:   pointy.Float64(5.0),
+			wantTbbBase:  5,
+			wantTbbQuote: 10,
+		},
+		{
+			name:         "12. otb = 0 && tbb = 0; projected = cap",
+			cap:          10.0,
+			otb:          0,
+			tbb:          0,
+			inputPrice:   2.0,
+			inputAmount:  10.0,
+			wantPrice:    pointy.Float64(2.0),
+			wantAmount:   pointy.Float64(10.0),
+			wantTbbBase:  10,
+			wantTbbQuote: 20,
+		},
+		{
+			// note that in this case, newAmount >= 0, since newAmount = cap - otb - tbb
+			name:         "13. otb = 0 && tbb = 0; projected > cap, newAmount > 0",
+			cap:          10.0,
+			otb:          0,
+			tbb:          0,
+			inputPrice:   2.0,
+			inputAmount:  15.0,
+			wantPrice:    pointy.Float64(2.0),
+			wantAmount:   pointy.Float64(10.0),
+			wantTbbBase:  10,
+			wantTbbQuote: 20,
+		},
+		{
+			name:         "14. otb = 0 && tbb = 0; projected > cap, newAmount = 0",
+			cap:          0.0,
+			otb:          0,
+			tbb:          0,
+			inputPrice:   2.0,
+			inputAmount:  15.0,
+			wantPrice:    nil,
+			wantAmount:   nil,
+			wantTbbBase:  0,
+			wantTbbQuote: 0,
+		},
+		{
+			name:         "15. otb > 0 && tbb > 0; projected < cap, newAmount = 0",
```

Comment:
there may be an error with the name here, newAmount is not 0 in this test case.

I'm wondering if it was leftover from copy-pasting of a previous case, can we rename this to remove `newAmount = 0`?

Reply:
Thanks for catching - just removed `newAmount = 0`.